react/features/recording/components/LiveStream/StartLiveStreamDialog.web.js

```diff
@@ -341,6 +338,42 @@ class StartLiveStreamDialog extends Component {
             });
     }
 
+    /**
+     * Takes in a list of broadcasts from the YouTube API, removes dupes,
+     * removes broadcasts that cannot get a stream key, and parses the
+     * broacasts into flat objects.
+     *
+     * @param {Array} broadcasts - Broadcast descriptions as obtains from
+     * calling the YouTube API.
+     * @private
+     * @returns {Array} An array of objects that are flat and contain only data
+     * that is needed. The objects have the shape: {{
+     *     boundStreamID: string
+     *     status: string,
+     *     title: string
+     * }}
+     */
+    _parseBroadcasts(broadcasts) {
+        const parsedIds = new Set();
+        const parsedBroadcasts = [];
```

Comment:
can this not be re-used to check for already-added broadcasts (instead of having to use the additional set)?  it's not necessarily that having the additional set is so terrible in itself, just that we have a split source of truth.  another option would be to use a map of `broadcastId` -> `broadcastInfo (status & title)`

Reply:
I guess I didn't want to iterate over the array each time to look for objects with the same id, even though the cost penalty is really low. However, I can just make it an Object, with keys being the IDs, and return the object's values at the end.